src/api/v2/types.rs

```diff
@@ -193,44 +200,231 @@ impl Reply for Status {
 	}
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash)]
 #[serde(rename_all = "kebab-case")]
-pub enum Topics {
+pub enum Topic {
 	HeaderVerified,
 	ConfidenceAchieved,
 	DataVerified,
 }
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
 #[serde(rename_all = "kebab-case")]
-pub enum DataFields {
+pub enum DataField {
 	Data,
 	Extrinsic,
 }
 
-#[derive(Serialize, Deserialize, PartialEq)]
+#[derive(Serialize, Deserialize, PartialEq, Default)]
 pub struct Subscription {
-	pub topics: HashSet<Topics>,
-	pub data_fields: HashSet<DataFields>,
+	pub topics: HashSet<Topic>,
+	pub data_fields: HashSet<DataField>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct HeaderMessage {
+	block_number: u32,
+	header: Header,
+}
+
+impl TryFrom<avail_subxt::primitives::Header> for HeaderMessage {
+	type Error = anyhow::Error;
+
+	fn try_from(header: avail_subxt::primitives::Header) -> Result<Self, Self::Error> {
+		let header: Header = header.try_into()?;
+		Ok(Self {
+			block_number: header.number,
+			header,
+		})
+	}
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct Header {
+	hash: H256,
+	parent_hash: H256,
+	pub number: u32,
+	state_root: H256,
+	extrinsics_root: H256,
+	extension: Extension,
+}
+
+#[derive(Debug, Clone)]
+struct Commitment([u8; config::COMMITMENT_SIZE]);
+
+impl Serialize for Commitment {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: Serializer,
+	{
+		let hex_string = format!("0x{}", hex::encode(self.0));
+		serializer.serialize_str(&hex_string)
+	}
+}
+
+impl<'de> Deserialize<'de> for Commitment {
+	fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		const LEN: usize = config::COMMITMENT_SIZE * 2 + 2;
```

Comment:
Should we maybe explain the magic number(s) here

Reply:
Done.